ledger/src/blockstore_processor.rs

```diff
@@ -193,6 +206,8 @@ fn execute_batch(
         );
     }
 
+    prioritization_fee_cache.update(bank.clone(), executed_transactions.into_iter());
```

Comment:
`PrioritizationFeeCache::update` already performs some filtering (vote, 0-compute-unit, etc), what's the reasoning behind doing filtering before passing? Are there cases where we want it to update on non-executed txs?

Reply:
yea, `PrioritizationFeeCache` does some sanity checking on individual transactions. The main motivation here is to reduce possible manipulations on `PrioritizationFeeCache` by restricting only executed (and paid) transactions can update it.

Reply:
I get the motiviation and think that is the correct path. It's just unclear why some restrictions are done inside vs outside of `PrioritizationFeeCache::update`.

If we always want this restriction, it seems safer change params and place it inside.

Reply:
Ah I see, you mean this: `prioritization_fee_cache.update(bank, transactions, execution_result)` ? It's doable, tho I wonder if it is necessary to expose prioritization_fee_cache to "exeuction_result", which is a knowledge restrictively belong to execution path. My thought is to update p.fee cache for each block, the only necessary info are bank and tx. Cache itself sanitizes/checks input before using them, and callers are responsible for feeding proper txs to it. To me, these are two different type of responsibilities, wdyt?